msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/JsonHelper.java

```diff
@@ -40,15 +41,77 @@ static <T> T convertJsonToObject(final String json, final Class<T> tClass) {
     }
 
     static IdToken createIdTokenFromEncodedTokenString(String token) {
-        String idTokenJson;
+        return JsonHelper.convertJsonToObject(getTokenPayloadClaims(token), IdToken.class);
+    }
+
+    static String getTokenPayloadClaims(String token) {
         try {
-            idTokenJson = new String(Base64.getUrlDecoder().decode(token.split("\\.")[1]), StandardCharsets.UTF_8);
+            return new String(Base64.getUrlDecoder().decode(token.split("\\.")[1]), StandardCharsets.UTF_8);
         } catch (ArrayIndexOutOfBoundsException e) {
             throw new MsalClientException("Error parsing ID token, missing payload section.",
                     AuthenticationErrorCode.INVALID_JWT);
         }
+    }
+
+    //Converts a generic JSON string to a Map<String, Object> with relevant types
+    static Map<String, Object> parseJsonToMap(String jsonString) {
+        if (StringHelper.isBlank(jsonString)) {
+            return new HashMap<>();
+        }
+
+        try (JsonReader jsonReader = JsonProviders.createReader(jsonString)) {
+            jsonReader.nextToken();
+            return parseJsonObject(jsonReader);
+        } catch (IOException e) {
+            throw new MsalJsonParsingException(e.getMessage(), AuthenticationErrorCode.INVALID_JSON);
+        }
+    }
+
+    private static List<Object> parseJsonArray(JsonReader jsonReader) throws IOException {
```

Comment:
Same here - `readArray` might help you out?

Reply:
In the latest commit it now just uses readArray(JsonReader::readUntyped) rather than this method.